### ISSUE TITLE
VK_EXT_debug_utils not available in latest runtime

### ISSUE THREAD

**Issue by izoprene** [ValveSoftware/steam-runtime#124]
Hi,
I'm trying to use validation layers (VK_LAYER_LUNARG_standard_validation) with the latest Steam runtime but it appears that VK_EXT_debug_utils is not available when enumerating Instance Extensions ( vkEnumerateInstanceExtensionProperties() ). However VK_EXT_debug_report is being enumerated but that is a legacy extension.

I think this may be because the runtime's (Standard or beta) libvulkan-dev & libvulkan1 is at version 1.1.73. I sort of remember fixing this problem using the LunarG SDK version 1.1.85 but I'd rather not  change the Steam runtime Vulkan version as modifying it would defeat it's purpose.

Does the Valve Linux Steam team have a timeline for integrating a newer Vulkan SDK version?

**Comment by Plagman** [comment]
There isn't a firm timeline, but if it's useful and comes with a newer SDK we can look into it. An alternative in the short term is to locally unstall the newer SDK, since if the system has a newer version, the runtime scripts will consider it first. It would let you unblock yourself without modifying the runtime environment.

**Comment by izoprene** [comment]
Hi Plagman,
So if I understand correctly, installing the LunarG 1.1.85 SDK means that the runtime/run.sh would 'look outside' the Steam runtime directory and use the system Vulkan libraries (along with the VK_EXT_debug_utils extension)?

I'll will try this out, thanks.

**Comment by Plagman** [comment]
That's the idea; let me know if it works for you.

**Comment by izoprene** [comment]
I installed the LunarG 1.1.85 SDK (and built/installed SDL2.0.9 because Ubuntu 16.04 uses a non-Vulkan enabled version of SDL2) and had exactly the same problem.

It seems that on my system vkEnumerateInstanceExtensionProperties() only returns one Instance Extension: VK_EXT_debug_report, (not VK_EXT_debug_utils).

So I disabled checking that vkEnumerateInstanceExtensionProperties() is returning VK_EXT_debug_utils.

Instead I enumerate a list of Instance Extensions with SDL_Vulkan_GetInstanceExtensions() which returns: VK_KHR_surface & VK_KHR_xlib_surface.

Then I copy VK_KHR_surface, VK_KHR_xlib_surface and VK_EXT_debug_utils into an array[3] for my vulkanInstanceCreateInfo.ppEnabledExtensionNames and viola, the Instance is created and the debug messenger callback / validation works perfectly.

And building in the Steam runtime chroot and running with run.sh works as above with validation messenger callbacks, etc.

The Vulkan implementation for the Mesa driver for Skylake (my CPU) is incomplete so maybe it's a bug... 

Thanks again!
